### ISSUE TITLE
Weird behavior in Comm Links (Not user friendly)

### ISSUE THREAD

**Issue by samacumen** [mavlink/qgroundcontrol#4329]
Hi,

The issue I want to report is about creating a new connection via 'Comm Links' under Application Settings (1st Tab) on QGC.

I see that there is always a 'default UDP link' running and it figures out what the remote port should be for sending packets (which is kind of smart, but annoying from a user point of view). However, once I disconnect a new entry is automatically added. What I do not understand is why can't the user 'delete' the previous entries at will? Restart the QGC and all the previous entries disappear.

See the two images below, on creating new (automatic) connections and restarting QGC.

![1 new_connection_bug](https://cloud.githubusercontent.com/assets/1699546/21345906/5eb77c58-c6a2-11e6-8caa-ff4cdc27c845.png)
![2 after_restart_situation_normal](https://cloud.githubusercontent.com/assets/1699546/21345907/5eb81988-c6a2-11e6-92c7-f8a6e6030845.png)

2) Also, If a user creates an entry for connection, that should be taken up by default the next time. However, the QGC has its own logic behind to set up a UDP connection that it likes - kind of annoying!

Basically, this part of the interface is important and it should be made more user friendly.

Regards,
Shyam

**Comment by DonLakeFlyer** [comment]
This link is no longer shown in the list since it is an dynamic AutoConnect link. If you want to create your own UDP link with different settings than the standard UDP AutoConnect link you need to first turn off UDP AutoConnect from Settings/General page.

**Comment by samacumen** [comment]
Thanks @DonLakeFlyer . That answers part of my questions. Its good that we can disable the autoconnect.

However, when the autoconnect is made, there must be a dedicated link that user should not be able to disconnect, am I right? What is the point of disconnecting one and a new one (button) appears automagically below it - which is what is kind of annoying. Most users may still prefer the autoconnect. I know there was a long discussion on the forums regarding this sometime ago with Lorenz Meier and your team. I saw that some commits have been made to the QGC, does that fix any of these concerns?

Thanks for looking into the issue reported.

**Comment by DonLakeFlyer** [comment]
> What is the point of disconnecting one and a new one (button) appears automagically below it

That will no longer happen. The AutoConnect UDP link no longer shows up in the list so you can't disconnect it. The only way to disconnect the UDP AutoConnect link is by turning off AutoConnect for UDP in Settings/General.

**Comment by samacumen** [comment]
Ok, you mean to refer to "Default UDP Link" button no longer shows up in the list, am I right? Not the 'AutoConnect UDP link'.

Also, the issue I referred to was [https://github.com/mavlink/qgroundcontrol/issues/1681](url) kind of related to this. Is that also resolved?

**Comment by DonLakeFlyer** [comment]
> you mean to refer to "Default UDP Link" button no longer shows up in the list

Correct.

Not what you mean with respect to #1681. That is the discussion around the AutoConnect implementation which was completed ages ago.

**Comment by DonLakeFlyer** [comment]
@samacumen Do you have a 100% repro set of steps for this? I'll try some things, but easier if you already have it.
